cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaRebalanceAssemblyOperator.java

```diff
@@ -396,7 +397,14 @@ private Future<Void> reconcile(Reconciliation reconciliation, String host,
                                    CruiseControlApi apiClient, KafkaRebalance kafkaRebalance,
                                    KafkaRebalanceState currentState, KafkaRebalanceAnnotation rebalanceAnnotation) {
 
-        LOGGER.infoCr(reconciliation, "Rebalance action from state [{}]", currentState);
+
+        if (kafkaRebalance != null && kafkaRebalance.getStatus() != null
+                && Objects.equals(rebalanceStateConditionType(kafkaRebalance.getStatus()), "Ready")
```

Comment:
This would be maybe a bit more traditional and I guess it does the same?

```suggestion
                && "Ready".equals(rebalanceStateConditionType(kafkaRebalance.getStatus()))
```